app/services/product_tag_rules_filterer.rb

```diff
@@ -0,0 +1,111 @@
+# Takes a Spree::Variant AR object and filters results based on applicable tag rules.
+# Tag rules exists in the context of enterprise, customer, and variant_overrides,
+# and are applied to variant_overrides only. Returns a Spree::Variant AR object.
+
+class ProductTagRulesFilterer
+  def initialize(distributor, customer, variants_relation)
+    @distributor = distributor
+    @customer = customer
+    @variants_relation = variants_relation
+  end
+
+  def call
+    return variants_relation unless distributor_rules.any?
+
+    filter(variants_relation)
+  end
+
+  private
+
+  attr_accessor :distributor, :customer, :variants_relation
+
+  def distributor_rules
+    @distributor_rules ||= TagRule::FilterProducts.for(distributor).all
```

Comment:
This needs the `prioritised` scope, right?

Reply:
Added :+1: